proactive/ProActiveGateway.py

```diff
@@ -1,39 +1,64 @@
 import py4j
 import os
+from py4j.java_gateway import JavaGateway
+from py4j.java_collections import MapConverter
 
-from py4j.java_gateway import JavaGateway 
 
 
 class scheduler_gateway:
     
-    
-        
+
     """
     Simple client for the ProActive scheduler REST API
     See also https://try.activeeon.com/rest/doc/jaxrsdocs/overview-summary.html
     """
-    def __init__(self, base_url, username, password, credentials_path=None, insecure=True ):
+    def __init__(self, base_url,):
         """
         :param base_url: REST API base URL including host and port, for instance http://localhost:8080
         """
-        self.base_url = base_url
         ROOT_DIR = os.path.dirname(os.path.abspath(__file__))
         current_path = ROOT_DIR + "/java/lib/*"
-        gg = JavaGateway.launch_gateway(classpath=os.path.normpath(current_path), die_on_exit=True )
-        
+        self.base_url = base_url        
+        self.gateway = JavaGateway()               
+        self.gg = self.gateway.launch_gateway(classpath=os.path.normpath(current_path), die_on_exit=True )
+  
+
+    def connect(self,username, password, credentials_path=None, insecure=True ):
         credentials_file = None
         if credentials_path is not None:
-            credentials_file = gg.jvm.java.io.File(credentials_path)
-             
+            credentials_file = self.gg.jvm.java.io.File(credentials_path)
             
-        self.proactive_schduler_client = gg.jvm.org.ow2.proactive_grid_cloud_portal.smartproxy.RestSmartProxyImpl()
-        connection_info = gg.jvm.org.ow2.proactive.authentication.ConnectionInfo(base_url + "/rest", username, password, credentials_file, insecure)
+        self.proactive_schduler_client = self.gg.jvm.org.ow2.proactive_grid_cloud_portal.smartproxy.RestSmartProxyImpl()
+        connection_info = self.gg.jvm.org.ow2.proactive.authentication.ConnectionInfo(self.base_url + "/rest", username, password, credentials_file, insecure)
         self.proactive_schduler_client.init(connection_info);
-    
-    def submitFromCatalog(self, bucket_name, workflow_name):
         
-        return self.proactive_schduler_client.submitFromCatalog(self.base_url + "/catalog",bucket_name, workflow_name).getReadableName();
+    def disconnect(self):
+        self.proactive_schduler_client.disconnect()        
+    
```

Comment:
I understand that you restrict the scope somehow to the job level
in the implementations, we lack some functionalities:
- wait methods: like waitForJob, waitForAllJobs, waitForAnyJob.
- getting results: getTaskResult, getJobResult (here we can't skip the task level)
- kill, pause, resume, remove Job

Reply:
Yes, we can create other tasks to increase other functionalities